typed-protocols/src/Network/TypedProtocol/Driver.hs

```diff
@@ -91,20 +98,23 @@ instance Show (AnyMessage ps) => Show (TraceSendRecv ps) where
 -- Driver for normal peers
 --
 
--- | Run a peer with the given channel via the given codec.
---
--- This runs the peer to completion (if the protocol allows for termination).
---
-runPeer
-  :: forall ps (st :: ps) pr failure bytes m a .
-     (MonadThrow m, Exception failure)
+peerDriver
+  :: forall ps (st :: ps) pr failure err bytes m a .
+     ( MonadThrow m
+     , Exception err
+     )
   => Tracer m (TraceSendRecv ps)
   -> Codec ps failure m bytes
   -> Channel m bytes
+  -> (forall (st' :: ps) x .
+           PeerHasAgency (FlipAgency pr) st'
```

Comment:
Why `FlipAgency`? Shouldn't it be `forall pr`?

Reply:
Both will work; The `FlipAgency` is needed (at this is tighter assumption that `forall pr`), because this function will run under `Await` which flips the agency relative to the initial one.  I'll revert to `forall pr` since it's a weaker assumption, which makes it better suited for high-level API.